File: Java/Level1/로또의 최고 순위와 최저 순위.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int match = 0;
        int invalid = 0;
        Set<Integer> set = new HashSet<>();

        for (int win_num : win_nums)
            set.add(win_num);

        for (int lotto : lottos) {
            if (lotto == 0)
                invalid++;
            else if (set.contains(lotto))
                match++;
        }

        return new int[]{getRank(match + invalid), getRank(match)};
    }

    int getRank(int count) {
        return Math.min(6, 7 - count);
    }
}

File: programmers-coding-test/C++/Level1/로또의 최고 순위와 최저 순위.cpp
#include <string>
#include <vector>
#include <unordered_set>

using namespace std;

int getRank(int count) {
	return min(6, 7 - count);
}

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
	int match = 0;
	int invalid = 0;
	unordered_set<int> set(win_nums.begin(), win_nums.end());

	for (int lotto : lottos) {
		if (lotto == 0)
			invalid++;
		else if (set.find(lotto) != set.end())
			match++;
	}

	return vector<int>{getRank(match + invalid), getRank(match)};
}